### PR TITLE
README, tools/dm: add DM task config file intro

### PR DESCRIPTION
**Pull request by CaitinChen**
English doc: https://github.com/pingcap/docs/blob/master/tools/dm/task-configuration-file-intro.md
@csuzhangxc @lilin90 @GregoryIan PTAL

### CODE REVIEW

File: tools/dm/task-configuration-file-intro.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+# DM 任务配置文件介绍
+
+本文档主要介绍 Data Migration (DM) 的任务配置文件 [`task.yaml`](https://github.com/pingcap/dm/blob/master/dm/master/task.yaml)，包含[全局配置](#全局配置) 和[实例配置](#实例配置) 两部分。
+
+关于各配置项的功能和配置，请参阅[数据同步功能](/tools/dm/data-synchronization-features.md)。
+
+## 关键概念
+
+关于包括 `instance-id` 和 DM-worker ID 在内的关键概念的介绍，请参阅[关键概念](/tools/dm/dm-configuration-file-overview.md#关键概念)。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]: We change `instance-id` to `source-id` now.

File: tools/dm/task-configuration-file-intro.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+2. 根据全局配置编辑[实例配置](#实例配置)。
+
+## 全局配置
+
+### 基础信息配置
+
+```yaml
+name: test                      # 任务名称，需要全局唯一。
+task-mode: all                  # 任务模式，可设为 "full"、"incremental"、"all"。
+is-sharding: true               # 是否为分库分表任务。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
is-sharding: true               # 是否为分库分表合并任务。
```

File: tools/dm/task-configuration-file-intro.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+meta-schema: "dm_meta"          # 下游储存 `meta` 信息的数据库。
+remove-meta: false              # 是否在任务同步开始前移除上面的 `meta`（`checkpoint` 和 `onlineddl`）。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
remove-meta: false              # 是否在任务同步开始前移除该任务名对应的 `meta`（`checkpoint` 和 `onlineddl`）。
```
maybe more accurate.

File: tools/dm/task-configuration-file-intro.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    host: "192.168.0.1"
+    port: 4000
+    user: "root"
+    password: ""
+```
+
+`task-mode`
+
+- 描述：任务模式，可以通过任务模式来指定需要执行的数据迁移工作。
+- 值为字符串（`full`，`incremental` 或 `all`），默认值为 `all`。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]: no default value, remove *默认值为 `all`*?

File: tools/dm/task-configuration-file-intro.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    - `full`：只全量备份上游数据库，然后将数据全量导入到下游数据库。
+    - `incremental`：只通过 binlog 把上游数据库的增量修改同步到下游数据库, 可以设置实例配置的 `meta` 配置项来指定增量同步开始的位置。
+    - `all`：`full` + `incremental`。先全量备份上游数据库，将数据全量导入到下游数据库，然后从全量数据备份时导出的位置信息 (binlog position/GTID) 开始通过 binlog 增量同步数据到下游数据库。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
    - `all`：`full` + `incremental`。先全量备份上游数据库，将数据全量导入到下游数据库，然后从全量数据备份时导出的位置信息 (binlog position) 开始通过 binlog 增量同步数据到下游数据库。
```
in the current version, we do not use this GTID.

File: tools/dm/task-configuration-file-intro.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+
+filters:                                            # 上游数据库实例匹配的表的 binlog event 过滤规则集。
+    filter-rule-1:
+    ​    schema-pattern: "test_*"
+    ​    table-pattern: "t_*"
+    ​    events: ["truncate table", "drop table"]
+    ​    action: Ignore
+    filter-rule-2:
+        schema-pattern: "test_*"
+        events: ["All DML"]                         # 只执行 schema `test_*` 下面所有的 DML event。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
        events: ["all dml"]                         # 只执行 schema `test_*` 下面所有的 DML event。
```
use lower case.

File: tools/dm/task-configuration-file-intro.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+## 实例配置
+
+本小节定义具体的数据同步子任务，DM 支持从单个或者多个上游 MySQL 实例同步数据到同一个下游数据库实例。
+
+```
+mysql-instances:
+    -
+    ​    source-id: "mysql-replica-01"           # 上游实例或者复制组 ID，参考 `inventory.ini` 的 `source_id` 或者 `dm-master.toml` 的 `source-id` 配置。
+    ​    meta:                                   # 下游数据库的 `checkpoint` 不存在时 binlog 同步开始的位置; 如果 checkpoint 存在，则以 `checkpoint` 为准。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
    ​    meta:                                   # `task-mode` 为 `incremental` 且下游数据库的 `checkpoint` 不存在时 binlog 同步开始的位置; 如果 checkpoint 存在，则以 `checkpoint` 为准。
```

File: tools/dm/task-configuration-file-intro.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+    ​        binlog-name: binlog-00001
+    ​        binlog-pos: 4
+
+    ​    route-rules: ["route-rule-1", "route-rule-2"]    # 该上游数据库实例匹配的表到下游数据库的映射规则名称。
+    ​    filter-rules: ["filter-rule-1"]                  # 该上游数据库实例匹配的表的 binlog 过滤规则名称。
+    ​    column-mapping-rules: ["cm-rule-1"]              # 该上游数据库实例匹配的表的 column mapping 规则名称。
+    ​    black-white-list:  "bw-rule-1"                   # 该上游数据库实例匹配的表的 black and white list 过滤规则名称。
```
Comment (csuzhangxc) [inline tools/dm/task-configuration-file-intro.md]: unify these names in **功能配置集**? `black and white list` or `black white list`

File: tools/dm/task-configuration-file-intro.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+        expression: "partition id"
+        source-column: "id"
+        target-column: "id"
+        arguments: ["2", "test_", "t_"]
+
+mydumpers:                                          # mydumper 处理单元运行配置参数。
+    global:
+    ​    mydumper-path: "./mydumper"                 # mydumper binary 文件地址，这个无需设置，会由 Ansible 部署程序自动生成。
+    ​    threads: 16                                 # mydumper 从上游数据库实例导出数据的线程数量。
+    ​    chunk-filesize: 64                          # mydumper 生成的数据文件大小。
```
Comment (amyangfei) [inline tools/dm/task-configuration-file-intro.md]:
```suggestion
    ​    chunk-filesize: 64                          # mydumper 生成的数据文件大小，单位为 MB。
```

**Comment by lilin90** [comment]
@csuzhangxc @amyangfei Thanks! Addressed comments. PTAL again :)

**Pull request merged**